go/kbfs/libkbfs/block_tree.go

```diff
@@ -1054,16 +1071,30 @@ func (bt *blockTree) readyHelper(
 			err = bps.addNewBlock(
 				ctx, newInfo.BlockPointer, pb.pblock, readyBlockData, syncFunc)
 			if err != nil {
-				return nil, err
+				return err
 			}
 			err = bps.saveOldPtr(ctx, ptr)
 			if err != nil {
-				return nil, err
+				return err
 			}
 
 			parentPB.setChildBlockInfo(newInfo)
 			oldPtrs[newInfo] = ptr
-			newPtrs[newInfo.BlockPointer] = true
+			donePtrs[newInfo.BlockPointer] = true
+			return nil
+		}
+
+		pipeliner := pipeliner.NewPipeliner(maxParallelReadies)
+		for i := 0; i < len(pathsFromRoot); i++ {
+			err := pipeliner.WaitForRoom(ctx)
+			if err != nil {
+				return nil, err
+			}
+			go func(i int) { pipeliner.CompleteOne(worker(i)) }(i)
```

Comment:
I'd prefer an approach with a fixed number of goroutines iterating through the work, rather than launching one for each item. Just seems like unnecessary overhead.

Reply:
I was just trying out max's new "clean" way of doing things.  I think goroutine overhead is basically zero, but I don't mind doing it with explicit channels, so I pushed the suggested change (and rebased onto master).  Please take another look!

Reply:
Yeah I read Max's post, but it goes against the canonical suggestion in the [pipelines](https://blog.golang.org/pipelines) post, which is to fan-in to and fan-out from a pool of goroutines. The difference from threads is that you can have hundreds or thousands instead of tens.

At the very least, goroutines add ~4kB of memory overhead each.

Reply:
Sure, but the pipeliner way has the same number of concurrent goroutines as the new one, so memory overhead is the same.  (Moving the stackpointer back and forth for each new goroutine doesn't seem like it would matter.)

Reply:
> Moving the stackpointer back and forth for each new goroutine doesn't seem like it would matter.

Good point, I hadn't thought of it that way.